book/docs/writing-programs/compiling.mdx

```diff
@@ -109,4 +109,23 @@ fn main() {
     };
     build_program_with_args("../program", &args);
 }
-```
\ No newline at end of file
+```
+
+## Troubleshooting
+
+### C-bindings Compilation Issues
+
+When working with C-bindings in SP1, you might encounter compilation issues. One common solution is to use `clang` as your C compiler. Here's how to set it up:
+
+1. Make sure you have an up-to-date version of `clang` installed on your system
+2. Set the following environment variables before building:
+   ```bash
+   export CC=clang
```

Comment:
Actually it's wrong. since the `CC` env value is used internally in sp1 compiler, so it should point to a riscv32 compiler, which is unlikely the `clang` of the native system.

So I'd suggest to change it to smth like (for mac):

```bash
brew tap riscv-software-src/riscv 
brew install riscv-gnu-toolchain
export CFLAGS="-I/opt/homebrew/Cellar/riscv-gnu-toolchain/main/riscv64-unknown-elf/include
```

This way it'll point to all the headers of the target arch. 

The more profound solution though is to add proper headers into `sp1 -c` distribution.